library/alloc/tests/lib.rs

```diff
@@ -47,6 +47,7 @@
 #![feature(strict_provenance)]
 #![feature(once_cell)]
 #![feature(drain_keep_rest)]
+#![deny(fuzzy_provenance_casts)]
```

Comment:
A little surprised this is the only lint we turn on in the liballoc tests, but I'll look into that later.

Reply:
Yeah me too... I briefly tried denying `unsafe_op_in_unsafe_fn` and got too many red lines to pursue this further right now. ;)